frontend/src/pages/CourseDashboard/index.js

```diff
@@ -0,0 +1,218 @@
+  withSpaces: false,
+});
+
+const FILTER_KEY = 'regional-resources-dashboard-filters';
+
+export default function CourseDashboard() {
+  const { user } = useContext(UserContext);
+  const ariaLiveContext = useContext(AriaLiveContext);
+  const regions = allRegionsUserHasPermissionTo(user);
+  const defaultRegion = user.homeRegionId || regions[0] || 0;
+  const allRegionsFilters = useMemo(() => buildDefaultRegionFilters(regions), [regions]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [courseData, setCourseData] = useState({});
+  const [error, updateError] = useState();
+  const [resetPagination, setResetPagination] = useState(false);
+
+  const hasCentralOffice = useMemo(() => (
+    user && user.homeRegionId && user.homeRegionId === 14
+  ), [user]);
+
+  const getFiltersWithAllRegions = () => {
+    const filtersWithAllRegions = [...allRegionsFilters];
+    return filtersWithAllRegions;
+  };
+  const centralOfficeWithAllRegionFilters = getFiltersWithAllRegions();
+
+  const defaultFilters = useMemo(() => {
+    if (hasCentralOffice) {
+      return [...centralOfficeWithAllRegionFilters,
+        {
+          id: uuidv4(),
+          topic: 'startDate',
+          condition: 'is within',
+          query: defaultDate,
+        }];
+    }
+
+    return [
+      {
+        id: uuidv4(),
+        topic: 'region',
+        condition: 'is',
+        query: defaultRegion,
+      },
+      {
+        id: uuidv4(),
+        topic: 'startDate',
+        condition: 'is within',
+        query: defaultDate,
+      },
+    ];
+  }, [defaultRegion, hasCentralOffice, centralOfficeWithAllRegionFilters]);
+
+  const [filters, setFiltersInHook] = useSessionFiltersAndReflectInUrl(
+    FILTER_KEY,
+    defaultFilters,
+  );
+
+  const setFilters = useCallback((newFilters) => {
+    setFiltersInHook(newFilters);
+    setResetPagination(true);
+  }, [setFiltersInHook]);
+
+  // Remove Filters.
+  const onRemoveFilter = (id, addBackDefaultRegions) => {
+    const newFilters = [...filters];
+    const index = newFilters.findIndex((item) => item.id === id);
+    if (index !== -1) {
+      newFilters.splice(index, 1);
+      if (addBackDefaultRegions) {
+        // We always want the regions to appear in the URL.
+        setFilters([...allRegionsFilters, ...newFilters]);
+      } else {
+        setFilters(newFilters);
+      }
+    }
+  };
+
+  // Apply filters.
+  const onApplyFilters = (newFilters, addBackDefaultRegions) => {
+    if (addBackDefaultRegions) {
+      // We always want the regions to appear in the URL.
+      setFilters([
+        ...allRegionsFilters,
+        ...newFilters,
+      ]);
+    } else {
+      setFilters([
+        ...newFilters,
+      ]);
+    }
+
+    ariaLiveContext.announce(`${newFilters.length} filter${newFilters.length !== 1 ? 's' : ''} applied to courses with activity reports `);
+  };
+
+  const filtersToApply = useMemo(() => expandFilters(filters), [filters]);
```

Comment:
There is a hook that replaces all of this: useFilters. We should reuse existing code if possible. Check the regionalDashboard for something to copy paste from